mbbsd/register.c

```diff
@@ -280,23 +304,35 @@ delregcodefile(void)
 // Justify Utilities
 ////////////////////////////////////////////////////////////////////////////
 
+static void email_regcode(const char *regcode, const char *email)
+{
+    char buf[256];
+
+    /*
+     * It is intended to use BBSENAME instead of BBSNAME here.
+     * Because recently many poor users with poor mail clients
+     * (or evil mail servers) cannot handle/decode Chinese
+     * subjects (BBSNAME) correctly, so we'd like to use
+     * BBSENAME here to prevent subject being messed up.
+     * And please keep BBSENAME short or it may be truncated
+     * by evil mail servers.
+     */
+    snprintf(buf, sizeof(buf), " " BBSENAME " - [ %s ]", regcode);
+
+    bsmtp("etc/registermail", buf, email, "non-exist");
+}
+
 static void
 email_justify(const userec_t *muser)
 {
-	char buf[256], genbuf[256];
-	/*
-	 * It is intended to use BBSENAME instead of BBSNAME here.
-	 * Because recently many poor users with poor mail clients
-	 * (or evil mail servers) cannot handle/decode Chinese
-	 * subjects (BBSNAME) correctly, so we'd like to use
-	 * BBSENAME here to prevent subject being messed up.
-	 * And please keep BBSENAME short or it may be truncated
-	 * by evil mail servers.
-	 */
-	snprintf(buf, sizeof(buf),
-		 " " BBSENAME " - [ %s ]", makeregcode(genbuf));
-
-	bsmtp("etc/registermail", buf, muser->email, "non-exist");
+	char regcode[256];
```

Comment:
Use a macro. No need for such a large buffer.

Reply:
Done.